lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,125 +325,172 @@ def create_launch_config(connection, module):
-                     result=result)
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(return_object))
 
 
 def delete_launch_config(connection, module):
-    name = module.params.get('name')
-    launch_configs = connection.get_all_launch_configurations(names=[name])
-    if launch_configs:
-        launch_configs[0].delete()
-        module.exit_json(changed=True)
-    else:
-        module.exit_json(changed=False)
+    try:
+        name = module.params.get('name')
+        launch_configs = connection.describe_launch_configurations(
+            LaunchConfigurationNames=[name]).get('LaunchConfigurations')
+        if launch_configs and len(launch_configs) > 0:
+            connection.delete_launch_configuration(
+                LaunchConfigurationName=launch_configs[0].get(
+                    'LaunchConfigurationName'))
+            module.exit_json(changed=True)
+        else:
+            module.exit_json(changed=False)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
 
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
-            name=dict(required=True, type='str'),
-            image_id=dict(type='str'),
-            key_name=dict(type='str'),
+            name=dict(required=True),
+            image_id=dict(),
+            instance_id=dict(),
+            key_name=dict(),
             security_groups=dict(type='list'),
-            user_data=dict(type='str'),
+            user_data=dict(),
             user_data_path=dict(type='path'),
-            kernel_id=dict(type='str'),
+            kernel_id=dict(),
             volumes=dict(type='list'),
-            instance_type=dict(type='str'),
+            instance_type=dict(),
             state=dict(default='present', choices=['present', 'absent']),
             spot_price=dict(type='float'),
-            ramdisk_id=dict(type='str'),
-            instance_profile_name=dict(type='str'),
+            ramdisk_id=dict(),
+            instance_profile_name=dict(),
             ebs_optimized=dict(default=False, type='bool'),
             associate_public_ip_address=dict(type='bool'),
             instance_monitoring=dict(default=False, type='bool'),
+            advanced_instance_monitoring=dict(default=False, type='bool'),
             assign_public_ip=dict(type='bool'),
             classic_link_vpc_security_groups=dict(type='list'),
-            classic_link_vpc_id=dict(type='str'),
-            vpc_id=dict(type='str')
+            classic_link_vpc_id=dict(),
+            vpc_id=dict(),
+            placement_tenancy=dict(default='default',
+                                   choices=['default', 'dedicated'])
         )
     )
 
     module = AnsibleModule(
         argument_spec=argument_spec,
-        mutually_exclusive = [['user_data', 'user_data_path']]
+        mutually_exclusive=[['user_data', 'user_data_path']]
     )
 
-    if not HAS_BOTO:
```

Comment:
No, you still have to check `HAS_BOTO3` (imported [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/cloudfront_facts.py#L229) and checked [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/cloudfront_facts.py#L558) )